### PR TITLE
Add models for `git-info` files

### PR DESCRIPTION
**Pull request by premun**
The tarball and the VMR contain so-called `git-info` files which are small XMLs where we store what revision of each individual repository the VMR was built from.
There is also another file called `AllRepoVersions.prop` which contains a list of all of the SHAs / versions extracted for easier manipulation.

Previously, these files were created here: https://github.com/dotnet/installer/blob/b2beceb5c62113a5600855d9665ac7f7c683cd15/src/SourceBuild/Arcade/src/Tarball_WriteSourceRepoProperties.cs

The VMR tooling also needs to create/read these so that it can initialize/update the VMR appropriately. This change will come as a follow-up. This PR only adds the models and the XML de/serialization.

https://github.com/dotnet/arcade/issues/10264

### CODE REVIEW

**Review by oleksandr-didyk — APPROVED** [review]

File: src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/VersionFiles.cs [hunk 1]
```diff
@@ -76,5 +79,63 @@ public static string CalculateDotnetToolsJsonElementName(string dependencyName)
...
+                {
+                    return ($"{releaseParts[1]}.{releaseParts[2]}", releaseParts[0]);
+                }
+                else if (int.TryParse(releaseParts[1], out int datePart) && int.TryParse(releaseParts[2], out int buildPart))
+                {
+                    if (datePart > 1 && datePart < 8 && buildPart > 1000 && buildPart < 10000)
+                    {
+                        return (releaseParts[2], $"{releaseParts[0]}.{releaseParts[1]}");
+                    }
+                    else
```
Comment (tkapin) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/VersionFiles.cs:120]: There is a return in the above branch, the else is not needed here.

File: src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/VersionFiles.cs [hunk 2]
```diff
@@ -76,5 +79,63 @@ public static string CalculateDotnetToolsJsonElementName(string dependencyName)
...
+            }
+
+            if (releaseParts.Length == 3)
+            {
+                // VSTest uses full dates for the first part of their preview build numbers
+                if (repoName.Contains("vstest"))
+                {
+                    return ($"{releaseParts[1]}.{releaseParts[2]}", releaseParts[0]);
+                }
+                else if (int.TryParse(releaseParts[1], out int datePart) && int.TryParse(releaseParts[2], out int buildPart))
```
Comment (tkapin) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/VersionFiles.cs:114]: The branching ends with this else-if without else so there might be some cases unhandled. Is this intentional?

**Review by tkapin — DISMISSED** [review]

File: src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/VersionFiles.cs (hunk 2 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/VersionFiles.cs:114]: Yes, this is intentional. The original code was a bit more convoluted:
https://github.com/dotnet/installer/blob/9ec5d7f1bc4202809d4c59407daead9c0c5df0c9/src/SourceBuild/Arcade/src/Tarball_WriteSourceRepoProperties.cs#L99 

but it is intended that if we fail to match some of the formats, we throw at the bottom.

Comment (premun) [reply src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/VersionFiles.cs:114]: I can make it an `if` only?

File: src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/VersionFiles.cs (hunk 1 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/VersionFiles.cs:120]: Ah, missed this when porting from https://github.com/dotnet/installer/blob/9ec5d7f1bc4202809d4c59407daead9c0c5df0c9/src/SourceBuild/Arcade/src/Tarball_WriteSourceRepoProperties.cs#L99

**Review by oleksandr-didyk — APPROVED** [review]

**Pull request merged**